src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs

```diff
@@ -954,6 +957,9 @@ private void ImportLdToken(int token)
             else
             {
                 Debug.Assert(obj is FieldDesc);
+                var field = (FieldDesc)obj;
```

Comment:
The assert is now of little value.

```suggestion
                var field = (FieldDesc)obj;
```